src/main/webapp/utilities/MapsInterface.js

```diff
@@ -0,0 +1,105 @@
+class MapInterface {
+    /**
+     * @param element the DOM element the map to be rendered
+     * @param centerCoordinates coordinates object that will be the center of the map
+     * @param options (optional) additional parameters that might be needed
+     */
+    constructor(element, centerCoordinates, options = {}) {
+        this.map = new google.maps.Map(element, {
+            center: centerCoordinates,
+            zoom: options.zoom || 8,
+        });
+
+        // Service that fetches the route (used on drawRoute)
+        this.directionsService = options.directionsService || new google.maps.DirectionsService();
+
+        // Service that draws a route on the map (used on drawRoute)
+        this.directionsRenderer = options.directionsRenderer || new google.maps.DirectionsRenderer({
+                suppressMarkers: true, // do not display default markers of the route
+            });
+
+        this.directionsRenderer.setMap(this.map);
+
+        this.markers = [];
+    }
+
+    /**
+     * @param coordinates coordinates object of the marker's position
+     * @param markerObject substitude object for marker (for testing)
```

Comment:
Why does the interface have an argument for testing? You can keep it if you want it but maybe flesh out what it does

Reply:
Added more descriptive documentation for those! We can completely eliminate this by decoupling implementation from interface (as discussed)! Added a TODO for that!